### ISSUE TITLE
Package Compatibility

### ISSUE THREAD

**Issue by christianjuth** [onejs/one#287]
This issue aims to outline all the package compatibility issues with One as One approaches it's stable release. 

I will update the issue as I become aware of other packages causing issues.

------

### Expo packages

Package | Description of Issue | Tested Version(s) | Reproduce Link | Fixed
-- | -- | -- | -- | --
expo-image | When I run `yarn dev` I get `"The JSX syntax extension is not currently enabled"` caused by expo using `.js` files instead of `.jsx`.
expo-blur | When I run `yarn dev` I get `"The JSX syntax extension is not currently enabled"` caused by expo using `.js` files instead of `.jsx`.

In theory, we should be able to fix these using One's Vite config deps ([see docs](https://onestack.dev/docs/configuration#deps)). Some messages on One Discord suggest there might be a second layer to this issue.

--------

### Misc
Package | Description of Issue | Tested Version(s) | Reproduce Link | Fixed
-- | -- | -- | -- | --
react-native-reanimated-carousel | `"x Build failed in 6.52s, error [vite:json] [plugin vite:json] virtual:rn-internals:react-native/package.json: Failed to parse JSON file."`

**Comment by zetavg** [comment]
`expo-blur` should work now with the fixes included in `1.1.341` 👍

**Comment by clayrisser** [comment]
Is there an exposed api that will allow people to get their own package support working without waiting for official support?

**Comment by clayrisser** [comment]
I guess that's what the deps config is for.

**Comment by natew** [comment]
`@lexical/react` should work now with `1.1.344`

**Comment by christianjuth** [comment]
`expo-image` was giving me trouble, but the following `vite.config.ts` fixed it.

```ts
export default {
  plugins: [
    one({
      // ...
    }),
    tamaguiPlugin({
      // ...
    }),
  ],
  resolve: {
    alias: {
      "expo-modules-core": path.resolve(
        __dirname,
        "node_modules/expo-modules-core/",
      ),
    },
  },
} satisfies UserConfig;

```

**Comment by SpaghettiC0des** [comment]
@natew should `one` patch `expo-linear-gradient` by default? I am doing this to make it work on the web.

```ts
      deps: {
        'expo-linear-gradient': {
          '**/*.js': ['flow', 'swc'],
        },
      },
```

**Comment by natew** [comment]
Yes i think we'd probably want that, likely we need to patch all expo module somehow, I want to set up a better way to do this internally automatically, but for now I'd accept a pr to `dePatches` doing this.

**Comment by SpaghettiC0des** [comment]
@natew I can submit a PR for this.

**Comment by SpaghettiC0des** [comment]
I think a separate issue to keep track of the patched expo modules would be better.

**Comment by shashwat2003** [comment]
`axios` throws this error on react-native:

![image](https://github.com/user-attachments/assets/80eb15e3-af4a-47c1-a0bb-7ebffc32a508)

adding  `util: true` to `one.deps` fixes it. Thanks to @natew

**Comment by rommyarb** [comment]
Will the new [Metro Mode](https://onestack.dev/docs/metro-mode) eliminate all these package compatibility issues?

**Comment by natew** [comment]
Update: Fixed expo-asset (#301) with a patch that properly re-exports functions from react-native's `resolveAssetSource` module. This is in commit 034762e1c and will be available in the next release.
